integration-tests/src/test/java/tachyon/client/concurrent/FileInStreamConcurrencyIntegrationTest.java

```diff
@@ -0,0 +1,101 @@
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.concurrent;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import com.google.common.collect.Lists;
+
+import tachyon.Constants;
+import tachyon.LocalTachyonClusterResource;
+import tachyon.client.ClientContext;
+import tachyon.client.StreamOptionUtils;
+import tachyon.client.TachyonFSTestUtils;
+import tachyon.client.file.FileInStream;
+import tachyon.client.file.TachyonFile;
+import tachyon.client.file.TachyonFileSystem;
+import tachyon.client.file.options.OutStreamOptions;
+import tachyon.conf.TachyonConf;
+import tachyon.exception.FileDoesNotExistException;
+import tachyon.exception.TachyonException;
+import tachyon.util.io.PathUtils;
+
+/**
+ * Tests the concurrency of {@link FileInStream}.
+ */
+public final class FileInStreamConcurrencyIntegrationTest {
+  private static final int BLOCK_SIZE = 30;
+
+  @ClassRule
+  public static LocalTachyonClusterResource sLocalTachyonClusterResource =
+      new LocalTachyonClusterResource(Constants.GB, Constants.KB, BLOCK_SIZE);
+  private static TachyonFileSystem sTfs = null;
+  private static TachyonConf sTachyonConf;
+  private static OutStreamOptions sWriteTachyon;
+
+  @BeforeClass
+  public static final void beforeClass() throws Exception {
+    sTfs = sLocalTachyonClusterResource.get().getClient();
+    sTachyonConf = sLocalTachyonClusterResource.get().getMasterTachyonConf();
+    sWriteTachyon = StreamOptionUtils.getOutStreamOptionsWriteTachyon(sTachyonConf);
+  }
+
+  /**
+   * Tests the concurrent read of {@link FileInStream}.
+   */
+  @Test
+  public void FileInStreamConcurrencyTest() throws Exception {
+    String uniqPath = PathUtils.uniqPath();
+    TachyonFile f =
+        TachyonFSTestUtils.createByteFile(sTfs, uniqPath, BLOCK_SIZE * 2, sWriteTachyon);
+
+    List<Thread> threads = Lists.newArrayList();
+    for (int i = 0; i < ClientContext.getConf().getInt(Constants.USER_BLOCK_MASTER_CLIENT_THREADS)
+        * 10; i ++) {
+      threads.add(new Thread(new FileRead(f)));
+    }
+
+    ConcurrencyTestUtils.assertConcurrent(threads, 100);
+  }
+
+  class FileRead implements Runnable {
+    private final TachyonFile mFile;
+
+    FileRead(TachyonFile file) {
+      mFile = file;
+    }
+
+    @Override
+    public void run() {
+      try {
+        FileInStream stream = sTfs.getInStream(mFile);
+        stream.read();
```

Comment:
I think this might not be necessary, because this is a concurrency test, and the correctness part is already verified in other integration/unit tests.